src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
+                    removedElement = (T) node.getItem();
+                    node.getPrev().setNext(node.getNext());
+                    last = node.getPrev();
+                } else {
+                    removedElement = (T) node.getItem();
+                    node.getPrev().setNext(node.getNext());
+                    node.getNext().setPrev(node.getPrev());
+                }
+                size--;
+                return removedElement;
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public boolean remove(T object) {
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (node.item == object || (node.item != null && node.item.equals(object))) {
+                if (i == 0 && size == 1) {
+                    first = null;
+                } else if (i == 0) {
+                    node.getNext().setPrev(node.getPrev());
+                    first = node.getNext();
+                } else if (i == size - 1) {
+                    node.getPrev().setNext(node.getNext());
+                    last = node.getPrev();
+                } else {
+                    node.getPrev().setNext(node.getNext());
+                    node.getNext().setPrev(node.getPrev());
+                }
+                size--;
+                return true;
+            }
+            node = node.getNext();
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void validateIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index provided");
+        }
+    }
+
+    private void validateAddIndex(int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index provided");
+        }
+    }
+
+    private static class Node<E> {
+        private E item;
+        private MyLinkedList.Node<E> next;
+        private MyLinkedList.Node<E> prev;
+
+        Node(E element) {
+            this.item = element;
+            this.next = null;
+            this.prev = null;
+        }
+
+        Node(MyLinkedList.Node<E> prev, E element, MyLinkedList.Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+
+        public E getItem() {
+            return item;
+        }
+
+        public void setItem(E item) {
+            this.item = item;
+        }
+
+        public Node<E> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<E> next) {
+            this.next = next;
+        }
+
+        public Node<E> getPrev() {
+            return prev;
+        }
+
+        public void setPrev(Node<E> prev) {
+            this.prev = prev;
+        }
```

Comment:
Why do you need to get and set methods for your nested class? Are you sure about this decision?